newsfragments/4241.misc.rst

```diff
@@ -0,0 +1 @@
+Improvements to `Path`-related type annotations when it could be ``str | PathLike`` -- by :user:`Avasam`
```

Comment:
I marked this as `misc` rather than `feature` because for now type-checking users still use `types-setuptools`